Contributor.md

```diff
@@ -0,0 +1,88 @@
+
+# Contributor Guidelines
+
+Thank you for considering contributing to our project! We welcome contributions of all kinds, including bug fixes, new features, documentation, and tests. Please follow the guidelines below to ensure a smooth contribution process.
+
+## How to Contribute
+
+### 1. Fork the Repository
+        Fork this repository to your GitHub account and clone it locally.
+
+        ``` 
+        git clone https://github.com/your-username/project-name.git
+        cd project-name 
+        ```
+
+### 2. Fork the Repository
+        Follow the project’s setup instructions in the `README.md` file to prepare your development environment.
+
+### 3. Create a Branch
+        Create a new branch for your contribution. Use a descriptive name.
+
+        ``` 
+        git checkout -b feature/your-feature-name 
+        ```
+
+### 4. Make Your Changes
+        Implement your changes, and ensure they adhere to the project's coding standards.
+
+### 5. Write Tests
+        Add or update tests to cover your changes, if applicable.
+
+### 6. Run Tests
+        Run all tests to ensure your changes do not break existing functionality.
+
+        ``` 
+        make test 
+        ```
+
+### 7. Commit Your Changes
+        Write a clear and concise commit message.
```

Comment:
Commit messages should addhere to conventional commits standard: https://www.conventionalcommits.org/en/v1.0.0/#specification

(the example below should be ammended to reflect this).

Maybe also highlight expected commit structure: each commit should contain a coherent logical change, a commit should depend only on earlier commits, fixes/changes/updates should be fixed up into existing commits rather than addded as new commits, etc.